docs/high-availability.md

```diff
@@ -0,0 +1,196 @@
+# High Availability Kubernetes Clusters
+
+## Introduction
+This document describes how to build a high-available Kubernetes cluster.  This is a fairly advanced topic.
+Users who merely want to experiment with Kubernetes are encouraged to use configurations that are simpler to set up such as
+the simple [Docker based single node cluster instructions](https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/getting-started-guides/docker.md),
+or try [Google Container Engine](https://cloud.google.com/container-engine/) for hosted Kubernetes.
+
+Also, at this time high availability support for Kubernetes is not continuously tested in our end-to-end (e2e) testing.  We will
+be working to add this continuous testing, but for now the single-node master installations are more heavily tested.
+
+## Overview
+Setting up a truly reliable, highly available distributed system requires a number of steps, it is akin to
+wearing underwear, pants, a belt, suspenders, another pair of underwear, and another pair of pants.  We go into each
+of these steps in detail, but a summary is given here to help guide and orient the user.
+
+The steps involved are as follows:
+   * Setting up a redundant, reliable storage layer
+   * Installing a health-checking daemon runner
+   * Turning up a clustered instance of the etcd key/value store
+   * Starting replicated, load balanced Kubernetes API servers
+   * Setting up master-elected Kubernetes scheduler and controller-manager daemons
+
+Here's what the system should look like when it's finished:
+![High availability Kubernetes diagram](high-availability/ha.png)
+
+Ready? Let's get started.
+
+## Initial set-up.
+The remainder of this guide assumes that you are setting up a 3-node clustered master, where each machine is running some flavor of Linux.
+Examples in the guide are given for Debian distributions, but they should be easily adaptable to other distributions.
+Likewise, this set up should work whether you are running in a public or private cloud provider, or if you are running
+on bare metal.
+
+The easiest way to implement an HA Kubernetes cluster is to start with an existing single-master cluster.  The
+instructions at [https://get.k8s.io](https://get.k8s.io)
+describe easy installation for single-master clusters on a variety of platforms.
+
+## Reliable storage
```

Comment:
this section could be simplified to roughly state: 0. kubernetes relies on etcd for reliable storage, make sure your comfortable w/ your etcd deployment; 1. though etcd upstream touches on this, you can use reliable volumes attached to your nodes (like GPD, Gluster, Ceph, etc) or ISCSI, RAID, etc.; 2. a pitfall you should avoid is...[putting etcd data in /tmp?]

but 1. and 2. could be optional.

also, the "clusterd etcd" section could be moved up and merged here.

imho, the story is "kube needs etcd" "make etcd ha (here's a link)" "if you want extra assurances use non-local reliable or RAIDed storage"